sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

```diff
@@ -19,6 +19,27 @@
 
 #include <cstdint>
 
+#ifndef __SYCL_DEVICE_ONLY__
+/// ESIMD_CPU Emulation support using esimd_cpu plugin
+
+/// Definition macro to be referenced in CM header files for
+/// preventing build failure caused by symbol conflicts between llvm
+/// and CM - e.g. vector.
+#define __SYCL_EXPLICIT_SIMD_PLUGIN__
+
+// Header files required for accessing CM-managed resources - image,
+// buffer, runtime API etc.
+namespace cm_support {
+#include <CL/cm_rt.h>
+} // namespace cm_support
+
+#include <CL/sycl/backend_types.hpp>
+#include <CL/sycl/detail/pi.hpp>
+#include <sycl/ext/intel/experimental/esimd/detail/atomic_intrin.hpp>
+#include <sycl/ext/intel/experimental/esimd/emu/detail/esimdcpu_device_interface.hpp>
```

Comment:
this line leads to the failure of esimd/odr.cpp test failure. `esimdcpu_device_interface.hpp` defines the `getESIMDDeviceInterface` function, and is included into two sources which are then linked.

You should either add `inline` to this function or move its definition to some .cpp file which gets linked into libsycl.so. I'd prefer the second. @romanovvlad - do you have a preference?

Reply:
@romanovvlad , what is your preference?

And, if a new `.cpp` files is to be created, where should it be placed? `$src/llvm/sycl/source/detail`? (where `pi.cpp` is located)

Reply:
The second option sounds better.

Reply:
Location & name of the new cpp file : `sycl/source/esimdcpu_device_interface.cpp`

(REF : Header `sycl/include/sycl/ext/intel/experimental/esimd/emu/detail/esimdcpu_device_interface.hpp`)